### DISCUSSION TITLE
Develop a PowerShell Module

### DISCUSSION THREAD

**Discussion by joerodgers** [spectreconsole/spectre.console#887] [Ideas]
Would love a PowerShell module that allows the creation of some basic tables without having to use the .NET code syntax. Did a small PoC, but it's a lot of code that could be simplified with some cmdlets.

https://gist.github.com/joerodgers/c08a911e545575d382b7d0625699ab08

**Comment by lloydjatkinson** [comment]
Hey, just wanted to say this would be really great to have official support for. PowerShell modules can be written in C#, so I bet getting Spectre loaded in a module would not be a herculean effort! Having any kind of progress and interaction, ironically, is very limited currently with PowerShell. It's all .NET so it feels like all the right parts are in place and just need bridging together! About the best you can do is this:

![image](https://user-images.githubusercontent.com/5633938/225081730-1662fbec-e109-4aa2-8472-9225dc8522b6.png)

Also as shown in the proof of concept you still need to get the DLL's which is yet another barrier for automations and scripts. Having a PS module on PS Gallery would be 🔥.

**Comment by ShaunLawrie** [comment]
I wrote this one which supports pretty much all of spectre consoles functionality now https://pwshspectreconsole.com/
